src/main/java/seedu/address/logic/commands/SearchPatientCommand.java

```diff
@@ -0,0 +1,197 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.TemperatureRange;
+
+
+/**
+ * Search a person or a list of person according to a name or a range of temperature.
+ */
+public class SearchPatientCommand extends Command {
+
+    public static final String COMMAND_WORD = "searchpatient";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Search a patient in Covigent. "
+            + "Parameters: "
+            + PREFIX_NAME + "NAME || "
+            + PREFIX_TEMP_RANGE + "TEMPERATURERANGE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe ";
+
+    public static final String MESSAGE_SEARCH_PERSON_SUCCESS = "Person found: %1$s";
+    public static final String MESSAGE_NOT_FOUND = "At least one field to edit must be provided.";
+    public static final String MESSAGE_PERSON_NOT_FOUND = "The person you entered is not in the list.";
+    public static final String MESSAGE_SEARCH_PERSON_LIST_SUCCESS = "Person match your criteria found: \n";
+
+    private final SearchPatientDescriptor searchPatientDescriptor;
+    private final int areaNotFound = 0;
+    private final int areaIsName = 1;
+    private final int areaIsTemperature = 2;
+    private final int tooManyArea = 3;
+
+    /**
+     * Constructs an SearchPatientCommand to edit the patient with the name {@code String}.
+     *
+     * @param searchPatientDescriptor Description of the searchPerson command.
+     */
+    public SearchPatientCommand(SearchPatientDescriptor searchPatientDescriptor) {
+        requireNonNull(searchPatientDescriptor);
+        this.searchPatientDescriptor = searchPatientDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        final int areaToSearch = this.confirmArea(searchPatientDescriptor);
+        ArrayList<Person> personWithinTemperatureRange = new ArrayList<>();
+        List<Person> personList = model.getFilteredPersonList();
+
+        if (areaToSearch == areaNotFound) {
+            throw new CommandException(MESSAGE_NOT_FOUND);
+        }
+
+        else if (areaToSearch == areaIsName) {
+            String nameToSearch = searchPatientDescriptor.getName().toString().trim().toLowerCase();
+            Person personFound = personList.get(0);
+            boolean isPersonFound = false;
+
+            for (Person person : personList) {
+                String personName = person.getName().toString().trim().toLowerCase();
+                if (personName.equals(nameToSearch)) {
+                    personFound = person;
+                    isPersonFound = true;
```

Comment:
Actually you can remove the isPersonFound attribute totally. if (personName.equals(nameToSearch) can just return new CommandResult(String.format(MESSAGE_SEARCH_PERSON_SUCCESS, personFound)) directly. If the execution of code proceeds beyond this for loop, can just throw new CommandException outside. Just a suggestion to keep your code shorter :)

Reply:
sure!